src/components/linterlib/lacheck.ts

```diff
@@ -0,0 +1,141 @@
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs'
+
+import type { Extension } from '../../main'
+import { LinterUtil } from './linterutil'
+import { convertFilenameEncoding } from '../../utils/convertfilename'
+
+export class LaCheck {
+    readonly #linterName = 'LaCheck'
+    readonly linterDiagnostics: vscode.DiagnosticCollection = vscode.languages.createDiagnosticCollection(this.#linterName)
+    readonly #linterUtil: LinterUtil
+
+    constructor(private readonly extension: Extension) {
+        this.#linterUtil = new LinterUtil(extension)
+    }
+
+    async lintRootFile() {
+        this.extension.logger.addLogMessage('Linter for root file started.')
+        if (this.extension.manager.rootFile === undefined) {
+            this.extension.logger.addLogMessage('No root file found for linting.')
+            return
+        }
+
+        const filePath = this.extension.manager.rootFile
+        const configuration = vscode.workspace.getConfiguration('latex-workshop', vscode.Uri.file(filePath))
+        const command = configuration.get('linting.lacheck.exec.path') as string
+
+        let stdout: string
+        try {
+            stdout = await this.#linterUtil.processWrapper('root file', command, [filePath], {cwd: path.dirname(this.extension.manager.rootFile)})
+        } catch (err: any) {
+            if ('stdout' in err) {
+                stdout = err.stdout as string
+            } else {
+                return
+            }
+        }
+        this.parseLog(stdout)
+    }
+
+    async lintFile(document: vscode.TextDocument) {
```

Comment:
The methods `lintRootFile` and `lintFile` share a lot of common code. Actually, they only differ by a couple of lines. I suggest to refactor these two methods to extract the command part to ease maintainance.